javaslang-benchmark/src/test/java/javaslang/benchmark/SlangPriorityQueue.java

```diff
@@ -0,0 +1,99 @@
+package javaslang.benchmark;
+
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.*;
+import scala.compat.java8.functionConverterImpls.FromJavaBiFunction;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Warmup(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 3, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 2, jvmArgsAppend = { "-XX:+UseG1GC", "-Xss500m", "-Xms1g", "-Xmx1g", "-ea" })
```

Comment:
stack of 1/2 gb :D ... it seems that Scalaz doesn't use tail recursion, where it should have